src/services/google-analytics.ts

```diff
@@ -0,0 +1,75 @@
+import { VerificationFragment } from "@govtechsg/oa-verify";
+import { OpenAttestationDocument } from "@govtechsg/open-attestation";
+import { useEffect } from "react";
+import ReactGA from "react-ga4";
+
+const GTAG_ID = process.env.REACT_APP_GTAG_ID;
+
+export const useGoogleAnalytics = (): void => {
+  /* Initialise Google Analytics 4 if GTAG_ID is provided */
+  useEffect(() => {
```

Comment:
this will only capture the current page and not track navigation, is that the desired effect ?

Reply:
Yes, it is the desired effect. This is the initialisation which should only be called once.

The `ReactGA.send("pageview")` is explicitly called afterwards because the pageview event is not automatically triggered on first initialisation but will be automatically triggered by the GA script on subsequent page views.

This behaviour has been verified with the help of the [Google Tag Assistant](https://tagassistant.google.com).

Reply:
Maybe add this as a comment?